Recast/Source/RecastArea.cpp

```diff
@@ -513,79 +513,113 @@ void rcMarkConvexPolyArea(rcContext* context, const float* verts, const int numV
 
 int rcOffsetPoly(const float* verts, const int numVerts, const float offset, float* outVerts, const int maxOutVerts)
 {
-	// TODO (graham): Better variable names here.
 	const float MITER_LIMIT = 1.20f;
+    const float EPSILON = 1e-6f;
 
-	int n = 0;
+	int numOutVerts = 0;
 
-	for (int i = 0; i < numVerts; i++)
+	for (int vertIndex = 0; vertIndex < numVerts; vertIndex++)
 	{
-		const int a = (i + numVerts - 1) % numVerts;
-		const int b = i;
-		const int c = (i + 1) % numVerts;
-		const float* va = &verts[a * 3];
-		const float* vb = &verts[b * 3];
-		const float* vc = &verts[c * 3];
-		float dx0 = vb[0] - va[0];
-		float dy0 = vb[2] - va[2];
-		float d0 = dx0 * dx0 + dy0 * dy0;
-		if (d0 > 1e-6f)
+        // Grab three vertices of the polygon.
+		const int vertIndexA = (vertIndex + numVerts - 1) % numVerts;
+		const int vertIndexB = vertIndex;
+		const int vertIndexC = (vertIndex + 1) % numVerts;
+		const float* vertA = &verts[vertIndexA * 3];
+		const float* vertB = &verts[vertIndexB * 3];
+		const float* vertC = &verts[vertIndexC * 3];
+
+        // From A to B on the x/z plane
+		float ABdeltaX = vertB[0] - vertA[0];
```

Comment:
I think these names does not make things any more clear, some suggestions:
ABdeltaX/Y -> prevSegmentDirX/Y
BCdeltaX/Y -> curSegmentDirX/Y

ABperpX/Y -> prevSegmentNormX/Y
BCperpX/Y -> curSegmentNormX/Y

Reply:
Maybe add a helper function for the normalization, it's known concept but makes a lot of noise in this function.